packages/babel-types/src/definitions/flow.js

```diff
@@ -1,400 +1,422 @@
   },
 });
 
 defineType("InterfaceDeclaration", {
   visitor: ["id", "typeParameters", "extends", "body"],
   aliases: ["Flow", "FlowDeclaration", "Statement", "Declaration"],
   fields: {
-    // todo
+    id: validateType("Identifier"),
+    typeParameters: validateOptionalType("TypeParameterDeclaration"),
+    extends: validate(arrayOfType("InterfaceExtends")),
+    mixins: validate(arrayOfType("InterfaceExtends")),
+    body: validateType("ObjectTypeAnnotation"),
   },
 });
 
 defineType("IntersectionTypeAnnotation", {
   visitor: ["types"],
-  aliases: ["Flow"],
+  aliases: ["Flow", "FlowType"],
   fields: {
-    // todo
+    types: validate(arrayOfType("Flow")),
   },
 });
 
 defineType("MixedTypeAnnotation", {
-  aliases: ["Flow", "FlowBaseAnnotation"],
+  aliases: ["Flow", "FlowType", "FlowBaseAnnotation"],
 });
 
 defineType("EmptyTypeAnnotation", {
-  aliases: ["Flow", "FlowBaseAnnotation"],
+  aliases: ["Flow", "FlowType", "FlowBaseAnnotation"],
 });
 
 defineType("NullableTypeAnnotation", {
   visitor: ["typeAnnotation"],
-  aliases: ["Flow"],
+  aliases: ["Flow", "FlowType"],
   fields: {
-    // todo
+    typeAnnotation: validateType("Flow"),
   },
 });
 
 defineType("NumberLiteralTypeAnnotation", {
-  aliases: ["Flow"],
+  aliases: ["Flow", "FlowType"],
   fields: {
-    // todo
+    value: validate(assertValueType("number")),
   },
 });
 
 defineType("NumberTypeAnnotation", {
-  aliases: ["Flow", "FlowBaseAnnotation"],
-  fields: {
-    // todo
-  },
+  aliases: ["Flow", "FlowType", "FlowBaseAnnotation"],
 });
 
 defineType("ObjectTypeAnnotation", {
   visitor: ["properties", "indexers", "callProperties"],
-  aliases: ["Flow"],
+  aliases: ["Flow", "FlowType"],
   fields: {
-    // todo
+    properties: validate(
+      arrayOfType(["ObjectTypeProperty", "ObjectTypeSpreadProperty"]),
+    ),
+    indexers: validateOptional(arrayOfType("ObjectTypeIndexer")),
+    callProperties: validateOptional(arrayOfType("ObjectTypeCallProperty")),
+    exact: validate(assertValueType("boolean")),
   },
 });
 
 defineType("ObjectTypeCallProperty", {
   visitor: ["value"],
   aliases: ["Flow", "UserWhitespacable"],
   fields: {
-    // todo
+    value: validateType("Flow"),
+    static: validate(assertValueType("boolean")),
   },
 });
 
 defineType("ObjectTypeIndexer", {
-  visitor: ["id", "key", "value"],
+  visitor: ["id", "key", "value", "variance"],
   aliases: ["Flow", "UserWhitespacable"],
   fields: {
-    // todo
+    id: validateOptionalType("Identifier"),
+    key: validateType("Flow"),
+    value: validateType(["Flow", "Identifier"]),
+    static: validate(assertValueType("boolean")),
+    variance: validateOptionalType("Variance"),
   },
 });
 
 defineType("ObjectTypeProperty", {
-  visitor: ["key", "value"],
+  visitor: ["key", "value", "variance"],
   aliases: ["Flow", "UserWhitespacable"],
   fields: {
-    // todo
+    key: validateType("Identifier"),
+    value: validateType(["Flow", "Identifier"]),
```

Comment:
`FlowType`

Also, this can't be an `Identifier`: in `[x: A]: B`, `B` is a `GenericTypeAnnotation` (already included in `FlowType`)